service/operator/operator.go

```diff
@@ -0,0 +1,146 @@
+package operator
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/giantswarm/awstpr"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/framework"
+	"github.com/giantswarm/operatorkit/tpr"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+)
+
+// Config represents the configuration used to create a new service.
+type Config struct {
+	// Dependencies.
+	BackOff           backoff.BackOff
+	K8sClient         kubernetes.Interface
+	Logger            micrologger.Logger
+	OperatorFramework *framework.Framework
+}
+
+// DefaultConfig provides a default configuration to create a new service by
+// best effort.
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+		BackOff:           nil,
+		K8sClient:         nil,
+		Logger:            nil,
+		OperatorFramework: nil,
+	}
+}
+
+// Operator implements the reconciliation of custom objects.
+type Operator struct {
+	// Dependencies.
+	backOff           backoff.BackOff
+	logger            micrologger.Logger
+	operatorFramework *framework.Framework
+
+	// Internals.
+	bootOnce sync.Once
+	mutex    sync.Mutex
+	tpr      *tpr.TPR
+}
+
+// New creates a new configured service.
+func New(config Config) (*Operator, error) {
+	// Dependencies.
+	if config.BackOff == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.BackOff must not be empty")
+	}
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.K8sClient must not be empty")
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+	if config.OperatorFramework == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.OperatorFramework must not be empty")
+	}
+
+	var err error
+
+	var newTPR *tpr.TPR
+	{
+		c := tpr.DefaultConfig()
+
+		c.K8sClient = config.K8sClient
+		c.Logger = config.Logger
+
+		c.Description = awstpr.Description
+		c.Name = awstpr.Name
+		c.Version = awstpr.VersionV1
+
+		newTPR, err = tpr.New(c)
+		if err != nil {
+			return nil, microerror.Maskf(err, fmt.Sprintf("creating TPR util for %s", awstpr.Name))
+		}
+	}
```

Comment:
Current standard should be something like this: https://github.com/giantswarm/kvm-operator/blob/7bf46cd0c5130d104516c854c420d78b99117eef/service/operator/operator.go. E.g. we use the operatorkit informer and also inject the TPR now.